pkgs/by-name/mi/miktex/package.nix

```diff
@@ -0,0 +1,241 @@
+        Programs/MiKTeX/PackageManager/mpm/mpm.cpp \
+        Programs/MiKTeX/Yap/MFC/StdAfx.h \
+        Programs/MiKTeX/initexmf/initexmf.cpp \
+        Programs/MiKTeX/miktex/miktex.cpp \
+        --replace-fail "log4cxx/rollingfileappender.h" "log4cxx/rolling/rollingfileappender.h"
+
+      substitute cmake/modules/FindPOPPLER_QT5.cmake \
+        cmake/modules/FindPOPPLER_QT6.cmake \
+        --replace-fail "QT5" "QT6" \
+        --replace-fail "qt5" "qt6"
+
+      substituteInPlace Programs/TeXAndFriends/omega/otps/source/outocp.c \
+        --replace-fail 'fprintf(stderr, s);' 'fprintf(stderr, "%s", s);'
+    ''
+    # This patch fixes mismatch char types (signed int and unsigned int) on aarch64-linux platform.
+    # Should not be applied to other platforms otherwise the build will fail.
+    + lib.optionalString (stdenv.hostPlatform.isAarch64 && stdenv.hostPlatform.isLinux) ''
+      sed -i 's/--using-namespace=MiKTeX::TeXAndFriends/& --chars-are-unsigned/g' \
+        Programs/TeXAndFriends/Knuth/web/CMakeLists.txt
+    '';
+
+  nativeBuildInputs = [
+    bison
+    cmake
+    flex
+    fop
+    libxslt
+    pkg-config
+    kdePackages.wrapQtAppsHook
+    kdePackages.qttools
+    kdePackages.qt5compat
+    boost
+    dpkg
+  ];
+
+  buildInputs = [
+    apr
+    aprutil
+    bzip2
+    cairo
+    expat
+    fontconfig
+    freetype
+    fribidi
+    gd
+    gmp
+    graphite2
+    harfbuzzFull
+    hunspell
+    libjpeg
+    log4cxx
+    xz
+    mpfr
+    mpfi
+    libmspack
+    libressl
+    pixman
+    libpng
+    popt
+    uriparser
+    zziplib
+    kdePackages.poppler
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeBool "WITH_BOOTSTRAPPING" true)
+    (lib.cmakeBool "USE_SYSTEM_POPPLER" true)
+    (lib.cmakeBool "USE_SYSTEM_POPPLER_QT" true)
+    (lib.cmakeBool "MIKTEX_SELF_CONTAINED" false)
+    # Miktex infers install prefix by stripping CMAKE_INSTALL_BINDIR from the called program.
+    # It should not be set to absolute path in default cmakeFlags, otherwise an infinite loop will happen.
+    (lib.cmakeFeature "CMAKE_INSTALL_BINDIR" "bin")
+    (lib.cmakeFeature "CMAKE_INSTALL_LIBEXECDIR" "libexec")
+    (lib.cmakeFeature "MIKTEX_SYSTEM_LINK_TARGET_DIR" "${placeholder "out"}/bin")
+    (lib.cmakeFeature "MIKTEX_USER_LINK_TARGET_DIR" "${placeholder "out"}/bin")
+  ];
+
+  env = {
+    LANG = "C.UTF-8";
+    MIKTEX_REPOSITORY = "file://${miktexDebDist}/usr/share/miktex-texmf/miktex/bootstrapping/";
+  };
+
+  preBuild =
+    let
+      system_cpu =
+        if stdenv.hostPlatform.isDarwin then
+          stdenv.hostPlatform.darwinArch
+        else
+          stdenv.hostPlatform.parsed.cpu.name;
+      system_name = stdenv.hostPlatform.parsed.kernel.name;
+    in
+    ''
+      export HOME=$(mktemp -d)
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(pwd)/sandbox/miktex/bin/${system_name}-${system_cpu}
+
+      # high parallelism makes build unstable
+      if (( $NIX_BUILD_CORES > 4)); then
+        NIX_BUILD_CORES=4
+      fi
```

Comment:
Just do `enableParallelBuilding = false;`, if the race condition is there, it's always there regardless of the core count.